### PR TITLE
Add unit tests to validate bind failure messages.

### PR DESCRIPTION
**Pull request by szurbrigg**
- C++ code enforces supported argument/return types
- Some misc suggested comment cleanup

### CODE REVIEW

File: duktape/src/main/java/com/squareup/duktape/Duktape.java [hunk 1]
```diff
@@ -69,20 +69,22 @@ public String evaluate(String script) {
    * defines the interface implemented by {@code object} that will be accessible to JavaScript.
    * {@code type} must be an interface that does not extend any other interfaces, and cannot define
    * any overloaded methods.
+   *
+   * Methods of the interface may return void or any of the following supported argument types:
```
Comment (JakeWharton) [inline duktape/src/main/java/com/squareup/duktape/Duktape.java]: You need a `<p>` before this or Javadoc will collapse the sentence into the previous paragraph.

Comment (JakeWharton) [inline duktape/src/main/java/com/squareup/duktape/Duktape.java]: `{@code void}`

File: duktape/src/main/java/com/squareup/duktape/Duktape.java [hunk 2]
```diff
@@ -69,20 +69,22 @@ public String evaluate(String script) {
...
+   * boolean, int, double, java.lang.Boolean, java.lang.Integer, java.lang.Double, java.lang.String.
```
Comment (JakeWharton) [inline duktape/src/main/java/com/squareup/duktape/Duktape.java]: These should either be surrounded with `{@code ..}` or you can also use `{@link ..}` for them without the "java.lang." qualifier.

File: duktape/src/main/java/com/squareup/duktape/Duktape.java (hunk 1 above)
Comment (szurbrigg) [inline duktape/src/main/java/com/squareup/duktape/Duktape.java]: Done and done.

File: duktape/src/main/java/com/squareup/duktape/Duktape.java (hunk 2 above)
Comment (szurbrigg) [inline duktape/src/main/java/com/squareup/duktape/Duktape.java]: Nice.  Changed.

File: tests/src/androidTest/java/com/squareup/duktape/DuktapeTest.java [hunk 3]
```diff
@@ -284,4 +285,97 @@ public String getValue() {
...
+  }
+
+  @Test public void bindUnsupportedReturnType() {
+    try {
+      duktape.bind("value", UnsupportedReturnType.class, new UnsupportedReturnType() {
+        @Override public Date get() {
+          return null;
+        }
+      });
+      fail();
```
Comment (swankjesse) [inline tests/src/androidTest/java/com/squareup/duktape/DuktapeTest.java]: sweet

File: tests/src/androidTest/java/com/squareup/duktape/DuktapeTest.java [hunk 4]
```diff
@@ -284,4 +285,97 @@ public String getValue() {
...
+    try {
+      duktape.bind("illegal", UnsupportedArgumentType.class, new UnsupportedArgumentType() {
+        @Override public void set(Date d) {
+        }
+      });
+      fail();
+    } catch (IllegalArgumentException expected) {
+    }
+
+    // The state of our Duktape context is still valid, containing what was there before.
```
Comment (swankjesse) [inline tests/src/androidTest/java/com/squareup/duktape/DuktapeTest.java]: Nice test

**Pull request merged**